src/Datadog.Trace/Sampling/DefaultSamplingRule.cs

```diff
@@ -52,7 +57,11 @@ public void SetDefaultSampleRates(IEnumerable<KeyValuePair<string, float>> sampl
             {
                 foreach (var pair in sampleRates)
```

Comment:
If we change this method to take a `Dictionary<string, float>`, can we simply set the field instead of creating a new dictionary and copying the values? I would require changing a few method signatures (`RuleBasedSampler`, `ISampler`), but may be worth it.

```csharp
_sampleRates = sampleRates;
```

Reply:
Since I changed the type of the key with the latest commit, this is not possible anymore